### PR TITLE
fix-NXP-32215-Gatling-Feeder-randomly-failing

### PR DESCRIPTION
**Pull request by NourNuxeo**
## TL;DR
switched `no-appendfsync-on-rewrite` to `yes`.

---
### About `no-appendfsync-on-rewrite`
From [the Redis config file example](https://raw.githubusercontent.com/redis/redis/6.2/redis.conf)
```
# When the AOF fsync policy is set to always or everysec, and a background
# saving process (a background save or AOF log background rewriting) is
# performing a lot of I/O against the disk, in some Linux configurations
# Redis may block too long on the fsync() call. Note that there is no fix for
# this currently, as even performing fsync in a different thread will block
# our synchronous write(2) call.
#
# In order to mitigate this problem it's possible to use the following option
# that will prevent fsync() from being called in the main process while a
# BGSAVE or BGREWRITEAOF is in progress.
#
# This means that while another child is saving, the durability of Redis is
# the same as "appendfsync none". In practical terms, this means that it is
# possible to lose up to 30 seconds of log in the worst scenario (with the
# default Linux settings).
#
# If you have latency problems turn this to "yes". Otherwise leave it as
# "no" that is the safest pick from the point of view of durability.

no-appendfsync-on-rewrite no
```
### Testing
[Tested in staging](https://jenkins.platform-staging.dev.nuxeo.com/blue/organizations/jenkins/nuxeo%2Flts%2Fnuxeo-benchmark/detail/nuxeo-benchmark/12/pipeline). I have opened a shell in the redis container to assert the following:
- verified the presence of the config file.
- used `redis-cli` to check the configuration was taken into account
- collected the following logs with no more AOF:
```
1:M 06 Mar 2024 01:26:57.139 # Server initialized
1:M 06 Mar 2024 01:26:57.140 * Ready to accept connections
1:M 06 Mar 2024 01:28:15.869 * 10000 changes in 60 seconds. Saving...
1:M 06 Mar 2024 01:28:15.869 * Background saving started by pid 18
18:C 06 Mar 2024 01:28:15.906 * DB saved on disk
18:C 06 Mar 2024 01:28:15.906 * RDB: 0 MB of memory used by copy-on-write
1:M 06 Mar 2024 01:28:15.970 * Background saving terminated with success
...
```
old logs are like:
```
1:M 05 Mar 2024 12:40:29.946 # Server initialized
1:M 05 Mar 2024 12:40:29.947 * Ready to accept connections
1:M 05 Mar 2024 13:08:07.568 * Starting automatic rewriting of AOF on 6712387900% growth
1:M 05 Mar 2024 13:08:07.572 * Background append only file rewriting started by pid 39
1:M 05 Mar 2024 13:08:09.845 * AOF rewrite child asks to stop sending diffs.
39:C 05 Mar 2024 13:08:09.845 * Parent agreed to stop sending diffs. Finalizing AOF...
39:C 05 Mar 2024 13:08:09.845 * Concatenating 0.96 MB of AOF diff received from parent.
39:C 05 Mar 2024 13:08:09.853 * SYNC append only file rewrite performed
39:C 05 Mar 2024 13:08:09.856 * AOF rewrite: 7 MB of memory used by copy-on-write
1:M 05 Mar 2024 13:08:09.913 * Background AOF rewrite terminated with success
1:M 05 Mar 2024 13:08:09.913 * Residual parent diff successfully flushed to the rewritten AOF (0.04 MB)
1:M 05 Mar 2024 13:08:09.914 * Background AOF rewrite finished successfully
...
```

### Automation scripting breaking change
... Just kidding...

### CODE REVIEW

**Comment by bdelbosc** [comment]
Great finding, commit message could be improved, using conditional because it's correlation not a proved causality.
"NXP... Fix random Gatling Feeder crash

The AOF Redis backup process is probably the culprit, it's now disabled"
Also, PR comment is transient information to provide guidance for reviewer, any useful information must be in the NXP ticket.

**Comment by NourNuxeo** [comment]
Thank you I updated the description of the commit!
Sure I will update the ticket. Initially I wanted to ease the review process and ended up putting a lot here...

**Review by bdelbosc — APPROVED** [review]

File: charts/jenkins/pod-templates/nuxeo-benchmark-lts-2021.yaml.gotmpl [hunk 1]
```diff
@@ -32,7 +32,7 @@
     ttyEnabled: {{ .Values.podTemplates.ttyEnabled }}
     workingDir: {{ .Values.podTemplates.workingDir }}
   - args: ""
-    command: ""
+    command: "redis-server /usr/local/etc/redis/redis.conf"
```
Comment (ataillefer) [inline charts/jenkins/pod-templates/nuxeo-benchmark-lts-2021.yaml.gotmpl:35]: Do you need to alter the command? According to the [Configuration](https://github.com/bitnami/containers/tree/main/bitnami/redis#configuration-file) doc, mounting the conf file should be enough.

**Review by ataillefer — COMMENTED** [review]
In fact, can't we just use REDIS_AOF_ENABLED=no?
See the available [environment variables](https://github.com/bitnami/containers/tree/main/bitnami/redis#environment-variables).
Then, it could be set in the [redisEnvVars](https://github.com/nuxeo/platform-ci/blob/b24cfb55246095887dd825b5bcb7900eb63643f0/environments/common.yaml#L74).

File: charts/jenkins/pod-templates/nuxeo-benchmark-lts-2021.yaml.gotmpl (hunk 1 above)
Comment (NourNuxeo) [reply charts/jenkins/pod-templates/nuxeo-benchmark-lts-2021.yaml.gotmpl:35]: Sorry, I got confused as the repo only displays a main branch and the last commit was  "[bitnami/redis] Release 7.2.4-debian-12-r9". So I quickly jumped to conclusion... But looking it up in the 6.2 folder it is used indeed:
https://github.com/bitnami/containers/blob/23b5d6ecad3adc425b25b7a751c760e95678cae2/bitnami/redis/6.2/debian-12/rootfs/opt/bitnami/scripts/redis-env.sh#L31

Looking at the configuration, the path is a bit different. I need to change `etc` for `mounted-etc` in my PR if I'm correct ? 
And then no need for the command override.

Comment (ataillefer) [reply charts/jenkins/pod-templates/nuxeo-benchmark-lts-2021.yaml.gotmpl]: I think the path would be `/opt/bitnami/redis/mounted-etc/redis.con`, as stated in the Configuration doc.
But, IMO, if the environment variable works, it is simpler than a configuration file!

Comment (NourNuxeo) [reply charts/jenkins/pod-templates/nuxeo-benchmark-lts-2021.yaml.gotmpl]: It doesn't seem to be enough indeed.
So I will mount the new configMap on the default path so I don't have to override the command:
`/opt/bitnami/redis/mounted-etc/redis.conf`
Getting there 😄
